examples/todo/js/components/Root.js

```diff
@@ -1,70 +1,79 @@
+/* eslint-disable no-console */
+
 import React, { Component, PropTypes } from 'react';
-import io from 'socket.io-client';
 import { SubscriptionProvider } from 'relay-subscriptions';
+import io from 'socket.io-client';
 
 // most of this is implementation details. You can solve this however you want
 export default class Root extends Component {
   static propTypes = {
     environment: PropTypes.object.isRequired,
     children: PropTypes.element,
   };
-  _subscriptions = [];
+
   componentDidMount() {
     this._socket = io();
-    this._socket.on('connect', () => {
-      this.setState({ isConnected: true });
+
+    // receives subscription payloads
+    this._socket.on('subscription update', ({ id, data, errors }) => {
```

Comment:
Using a single channel for update means we don't have to dispose of listeners for the other channels.